mllib-dal/src/main/native/build.sh

```diff
@@ -18,5 +18,34 @@ if [[ $OAP_MLLIB_TESTING == "true" ]]; then
   exit 0
 fi
 
-make clean
-make -j
+if [[ -z $ONEDAL_VERSION ]]; then
+  if [[ -e $DAALROOT ]]; then
+    export ONEDAL_VERSION=$(echo "$DAALROOT" | awk -F '/' '{print $(NF)}')
+  elif [[ -e $DALROOT ]]; then
+    export ONEDAL_VERSION=$(echo "$DALROOT" | awk -F '/' '{print $(NF)}')
+  else
+    echo DAALROOT not defined!
+    exit 1
+  fi
+fi
+
+verlte() {
+    expr $(printf '%s\n%s' "$1" "$2" | sort -t '.' -k 1,1 -k 2,2 -k 3,3 -g | sed -n 2p) != "2023.2.0"
+}
+# Reference version
+reference_version="2023.2.0"
+
+# Compare versions
+verlte "$ONEDAL_VERSION" "$reference_version" && echo "1" || echo "0"
+result=$?
```

Comment:
use $? only get the exit code of the function. it can't get the echo output of the function.